CMakeLists.txt

```diff
@@ -4,77 +4,26 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 project(gbs LANGUAGES CXX)
 
-#
 # c++ conf
-#
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
-#
-# set output dirs (vs is a pain)
-#
-function(SET_OUT_DIR REQUIRED_ARG)
-    list(GET ARGV 0 TARGET_NAME)
-    list(GET ARGV 1 OUT_DIR)
-    message(${TARGET_NAME})
-    message(${OUT_DIR})
-    foreach( OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES} )
-        string( TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG )
-        set_property(TARGET ${TARGET_NAME} PROPERTY RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${OUT_DIR} )
-        set_property(TARGET ${TARGET_NAME} PROPERTY LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${OUT_DIR} )
-        set_property(TARGET ${TARGET_NAME} PROPERTY ARCHIVE_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${OUT_DIR} )
-    endforeach( OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES )
-endfunction()
-#
-# CODAN ENV
-#
-if(EXISTS $ENV{LIBRARY_PREFIX})
-    file(TO_CMAKE_PATH $ENV{LIBRARY_BIN} LIBRARY_BIN)
-    file(TO_CMAKE_PATH $ENV{LIBRARY_INC} LIBRARY_INC)
-    file(TO_CMAKE_PATH $ENV{LIBRARY_LIB} LIBRARY_LIB)
-elseif(EXISTS $ENV{CONDA_PREFIX})
-    file(TO_CMAKE_PATH $ENV{CONDA_PREFIX} CONDA_PREFIX)
-    file(TO_CMAKE_PATH ${CONDA_PREFIX}/Library/bin LIBRARY_BIN)
-    file(TO_CMAKE_PATH ${CONDA_PREFIX}/Library/include LIBRARY_INC)
-    file(TO_CMAKE_PATH ${CONDA_PREFIX}/Library/lib LIBRARY_LIB)
-else()
-    message(FATAL_ERROR "A conda environement is required")
-endif()
-
-message(${LIBRARY_LIB} ${LIBRARY_BIN} ${LIBRARY_INC})
-#
-# set install dir
-#
-function(SET_INSTALL_DIR REQUIRED_ARG)
-    list(GET ARGV 0 TARGET_NAME)
-    if(WIN32)
-        if(TARGET ${TARGET_NAME} )
-            install(TARGETS ${TARGET_NAME}
-                EXPORT  ${TARGET_NAME}Targets
-                COMPONENT python
-                LIBRARY  DESTINATION ${LIBRARY_LIB}
-                RUNTIME  DESTINATION ${LIBRARY_BIN}
-                INCLUDES DESTINATION ${LIBRARY_INC}
-                ARCHIVE  DESTINATION ${LIBRARY_LIB}
-            )
-            install(
-                EXPORT  ${TARGET_NAME}Targets
-                FILE ${TARGET_NAME}.cmake
-                NAMESPACE GBS::
-                DESTINATION ${LIBRARY_LIB}/cmake/${TARGET_NAME}
-            )
-        endif(TARGET ${TARGET_NAME} )
-    else() # TODO find export dir for linux
-            install(TARGETS ${TARGET_NAME}  
-            LIBRARY DESTINATION $ENV{LD_RUN_PATH}
-            )
-    endif()
+
+# Set installation directories (CMAKE_INSTALL_INCLUDEDIR, CMAKE_INSTALL_BINDIR, CMAKE_INSTALL_LIBDIR)
+include(GNUInstallDirs)
+
+message("-- CMake install prefix: " ${CMAKE_INSTALL_PREFIX})
```

Comment:
Depending if PREFIX env varaible is passed or not, to install correctly headers in the host env,  this modification is required:
```cmake
if($ENV{CONDA_BUILD})
    file(TO_CMAKE_PATH $ENV{PREFIX} ENV_PREFIX)
    set(CMAKE_INSTALL_PREFIX ${ENV_PREFIX}/Library)
    set(CMAKE_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_PREFIX}/include)
endif()
```

Reply:
I passed `LIBRARY_PREFIX` in the Windows build script `bld.bat` and created another `bld-dev.bat` meant for local builds that fill the env var with the active conda env.

IMHO this is preferable than modifying the CMakeLists files.

Reply:
good for me